internal/pkg/skuba/deployments/ssh/systemd.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ssh
+
+import (
+	"k8s.io/klog"
+)
+
+// IsServiceEnabled returns if a service is enabled
+func (t *Target) IsServiceEnabled(serviceName string) (bool, error) {
+	klog.V(1).Info("checking if skuba-update.timer is enabled")
+	if stdout, _, err := t.silentSsh("systemctl", "is-enabled", serviceName, "||", ":"); stdout == "enabled" {
```

Comment:
I don't like to check against a string. The string might change in the future or could be something completely different if the user has the system running in a language that is not English.

`t.silentSsh` is just a wrapper around [Session.Wait](https://godoc.org/golang.org/x/crypto/ssh#Session.Wait). According to the documentation:

> If the remote server does not send an exit status, an error of type *ExitMissingError is returned. If the command completes unsuccessfully or is interrupted by a signal, the error is of type *ExitError. Other error types may be returned for I/O problems.

I would instead look at the returned `error` and check if this is an instance of [*ExitError](https://godoc.org/golang.org/x/crypto/ssh#ExitError). If that's the case then look at the exit code of the remote command using the [ExitStatus](https://godoc.org/golang.org/x/crypto/ssh#Waitmsg.ExitStatus) method.

If the exit status == 1 -> the service was running, otherwise return the service is not running.

If the returned error is not of type `ExitStatus` then just propagate the error

Reply:
@flavio updated, thanks for the hint. please have another look